src/Arcus.Observability.Telemetry.Core/DependencyMeasurement.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using GuardNet;
+
+namespace Arcus.Observability.Telemetry.Core
+{
+    /// <summary>
+    /// Represents an instance to measure easily dependencies in an application.
+    /// </summary>
+    public class DependencyMeasurement : IDisposable
+    {
+        private readonly Stopwatch _stopwatch;
+        
+        private DependencyMeasurement(string dependencyData, DateTimeOffset startTime, Stopwatch stopwatch)
```

Comment:
Do we need stopwatch as input? We could just create a new one here

Reply:
Same with start time

Reply:
Hmm, is something like only do assignment in ctor. I guess that's a .NET guideline?
But is maybe fine by me.

Reply:
THat's how it is typically done in .NET yes